Content.Client/Popups/PopupSystem.cs

```diff
@@ -185,14 +184,58 @@ public override void FrameUpdate(float frameTime)
 
                 label.Visible = true;
             }
+
+            for (var i = _aliveCursorLabels.Count - 1; i >= 0; i--)
+            {
+                var label = _aliveCursorLabels[i];
+                if (label.TotalTime > PopupLifetime)
+                {
+                    label.Dispose();
+                    _aliveCursorLabels.RemoveAt(i);
+                }
+            }
```

Comment:
I have learnt that iterating forwards but using RemoveSwap(i) and then i-- also works and is faster

Reply:
Apparently iterating forwards might be quicker? I've never checked it myself but RemoveSwap is definitely a lot better anyway so.